File: backend/src/main/java/com/wellkorea/backend/invoice/domain/PaymentMethod.java
package com.wellkorea.backend.invoice.domain;

/**
 * Payment method enum for recording customer payments.
 */
public enum PaymentMethod {
    BANK_TRANSFER("계좌이체", "Bank Transfer"),
    CREDIT_CARD("카드결제", "Credit Card"),
    CHECK("수표", "Check"),
    CASH("현금", "Cash"),
    OTHER("기타", "Other");

    private final String labelKo;
    private final String labelEn;

    PaymentMethod(String labelKo, String labelEn) {
        this.labelKo = labelKo;
        this.labelEn = labelEn;
    }

    public String getLabelKo() {
        return labelKo;
    }

    public String getLabelEn() {
        return labelEn;
    }
}

File: frontend/src/entities/invoice/model/payment-method.ts
/**
 * Payment method enum and display configuration.
 */

export type PaymentMethod =
  | 'BANK_TRANSFER'
  | 'CREDIT_CARD'
  | 'CHECK'
  | 'CASH'
  | 'OTHER';

export interface PaymentMethodConfig {
  label: string;
  labelKo: string;
  icon: string;
}

export const paymentMethodConfig: Record<PaymentMethod, PaymentMethodConfig> = {
  BANK_TRANSFER: {
    label: 'Bank Transfer',
    labelKo: '계좌이체',
    icon: '🏦',
  },
  CREDIT_CARD: {
    label: 'Credit Card',
    labelKo: '신용카드',
    icon: '💳',
  },
  CHECK: {
    label: 'Check',
    labelKo: '수표',
    icon: '📝',
  },
  CASH: {
    label: 'Cash',
    labelKo: '현금',
    icon: '💵',
  },
  OTHER: {
    label: 'Other',
    labelKo: '기타',
    icon: '📋',
  },
};

/**
 * Get display label for payment method.
 */
export function getPaymentMethodLabel(
  method: PaymentMethod,
  korean = false
): string {
  return korean
    ? paymentMethodConfig[method].labelKo
    : paymentMethodConfig[method].label;
}

/**
 * Get all payment methods as options for select.
 */
export function getPaymentMethodOptions(
  korean = false
): Array<{ value: PaymentMethod; label: string }> {
  return (Object.keys(paymentMethodConfig) as PaymentMethod[]).map((key) => ({
    value: key,
    label: korean
      ? paymentMethodConfig[key].labelKo
      : paymentMethodConfig[key].label,
  }));
}
